### PR TITLE
Read all index data records in a chunk

### PR DESCRIPTION
**Pull request by liamembark**
Fixes: DATA-429

Description of Changes
======================
There are actually multiple INDEX_DATA records per chunk, so we just need to parse all of them instead of just the first!

Test Plan
=========
Extended the existing MessagesForTopic test to test all topics, not just the first.

### CODE REVIEW

**Comment by johnembark** [comment]
Awesome -- do you have a source/citation for where the data format is specified in documentation?

File: lib/embag.cc [hunk 1]
```diff
@@ -250,7 +249,7 @@ bool Bag::readRecords(boost::iostreams::stream<T> &stream) {
    * earlier in the file. Each CHUNK_INFO knows the position of its corresponding chunk.
    */
   for (size_t i = 0; i < chunk_count; i++) {
-    const auto info = chunk_infos_[i];
+    auto& info = chunk_infos_[i];
```
Comment (brendangeck) [inline lib/embag.cc:252]: What's the reasoning behind this change?

**Comment by brendangeck** [comment]
> Awesome -- do you have a source/citation for where the data format is specified in documentation?

Hey @johnembark 

I found this info in the docs here: http://wiki.ros.org/Bags/Format/2.0
Under the `Chunk info` section, I can see the following:
```
count | number of connections in the chunk | little-endian integer | 4 bytes
```

File: lib/embag.cc (hunk 1 above)
Comment (brendangeck) [reply lib/embag.cc:252]: Resolve offline: We need to update the info.num_messages value as we move through the info block

**Pull request merged**